src/connector/SpectodaWebBluetoothConnector.ts

```
@@ -531,202 +608,138 @@ export class WebBLEConnection {
         }
 
         // if any of these criteria are required, then we need to build a manufacturer data filter.
-        if (criterium.fwVersion || criterium.ownerSignature || criterium.productCode || criterium.adoptionFlag) {
+        if (criterium.fw || criterium.network || criterium.product || criterium.commisionable) {
           const company_identifier = 0x02e5; // Bluetooth SIG company identifier of Espressif
 
           delete filter.services;
 
-          let prefix = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
-          let mask = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+          const prefix = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)